### PR TITLE
Skip tests

### PR DESCRIPTION
**Pull request by reazulhoque**
This PR skips certain tests on some architecture.

### CODE REVIEW

**Comment by reazulhoque** [comment]
Tested with the following dependencies locally:

dpctl: d157ee0b1a9ba88d0f3b226bd069af2dd9a56b9a
Numba (patched): 864790061aa7548ac2b7ebcf448354def849e657
dpnp: 0.4.0
Result:

```
----------------------------------------------------------------------
Ran 157 tests in 71.137s
OK (skipped=2, expected failures=1)
```

File: numba_dppy/tests/test_numpy_math_functions.py [hunk 1]
```diff
@@ -4,6 +4,15 @@
 import dpctl
 import unittest
 
+def skip_tests(device_type):
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_numpy_math_functions.py:7]:
```suggestion
def is_gen12(device_type):
```

File: numba_dppy/tests/test_numpy_math_functions.py [hunk 2]
```diff
@@ -179,6 +188,7 @@ def f(a):
 
         self.assertTrue(np.all(c == -input_arr))
 
+    @unittest.skipIf(skip_tests("opencl:gpu"), "skipping test")
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_numpy_math_functions.py:191]:
```suggestion
    @unittest.skipIf(is_gen12("opencl:gpu"), "Gen12 not supported")
```

File: numba_dppy/tests/test_numpy_trigonomteric_functions.py [hunk 3]
```diff
@@ -4,6 +4,16 @@
...
+def skip_tests(device_type):
+    with dpctl.device_context(device_type):
+        q = dpctl.get_current_queue()
+        device = q.get_sycl_device()
+        name = device.get_device_name()
+        if "Gen12" in name:
+            return True
+
+        return False
+
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_numpy_trigonomteric_functions.py:16]: Duplication

**Review by PokhodenkoSA — APPROVED** [review]
Function could be named better. And one duplication. After fixing this PR could be merged.

File: numba_dppy/tests/test_numpy_math_functions.py (hunk 1 above)
Comment (reazulhoque) [reply numba_dppy/tests/test_numpy_math_functions.py]: Changed

File: numba_dppy/tests/test_numpy_math_functions.py (hunk 2 above)
Comment (reazulhoque) [reply numba_dppy/tests/test_numpy_math_functions.py]: Changed

File: numba_dppy/tests/test_numpy_trigonomteric_functions.py (hunk 3 above)
Comment (reazulhoque) [reply numba_dppy/tests/test_numpy_trigonomteric_functions.py]: Fixed

**Pull request merged**